src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -276,6 +276,24 @@ public void Remove(object key)
             StartScanForExpiredItemsIfNeeded(_options.Clock.UtcNow);
         }
 
+        /// <summary>
+        /// Removes all keys and values from the cache.
+        /// </summary>
+        public void Clear()
+        {
+            CheckDisposed();
+
+            var oldEntries = Interlocked.Exchange(ref _entries, new ConcurrentDictionary<object, CacheEntry>());
+            Interlocked.Exchange(ref _cacheSize, 0);
```

Comment:
This construct isn't thread safe (in total).
After `_entries` got updated, setting `_cacheSize` has a race-condition.

If this isn't a problem / doesn't matter, please add a comment stating this.